browser/src/layer/tile/CanvasTileLayer.js

```diff
@@ -2014,43 +2014,23 @@ L.CanvasTileLayer = L.Layer.extend({
 	},
 
 	_onCalcFunctionUsageMsg: function (textMsg) {
-		var pos = this._lastVisibleCursorRef._northEast;
-		this._map.uiManager.showFormulaTooltip(textMsg, pos);
+		var formulaAutocompletePopup = L.DomUtil.get('formulaautocompletePopup');
+		if (formulaAutocompletePopup)
+			this._map.fire('closeformulapopup');
+		this._map.fire('sendformulausagetext', {data: textMsg});
 	},
 
 	_onCalcFunctionListMsg: function (textMsg) {
 		if (textMsg.startsWith('hidetip')) {
-			this._map.uiManager.hideFormulaTooltip();
-		}
-		else {
-			var funcData = JSON.parse(textMsg);
-
-			if (window.mode.isMobile()) {
-				this._closeMobileWizard();
-
-				var data = {
-					id: 'funclist',
-					type: '',
-					text: _('Functions'),
-					enabled: true,
-					children: []
-				};
-
-				if (funcData.categories)
-					this._onCalcFunctionListWithCategories(funcData, data);
-				else
-					this._onCalcFunctionList(funcData, data);
-
-				if (funcData.wholeList)
-					this._map._functionWizardData = data;
-
-				this._openMobileWizard(data);
-			}
-			else {
-				var pos = this._lastVisibleCursorRef._northEast;
-				var tooltipinfo = this._getFunctionList(textMsg);
-				this._map.uiManager.showFormulaTooltip(tooltipinfo, pos);
-			}
+			var formulaAutocompletePopup = L.DomUtil.get('formulaautocompletePopup');
+			if (formulaAutocompletePopup)
+				this._map.fire('closeformulapopup');
```

Comment:
is that `if` needed? can we just fire event?

Reply:
Done with https://github.com/CollaboraOnline/online/pull/9201